db/collection.go

```diff
@@ -619,6 +619,17 @@ func (t *Txn) Discard() {
 	t.discarded = true
 }
 
+// RefreshCollection updates the transaction's collection reference from the master db map,
+// which may have received updates while the transaction is open.
+func (t *Txn) RefreshCollection() error {
```

Comment:
This is needed to pick up collection changes during a transaction.

Reply:
This seems odd given we're using a transaction here :P But I guess this actually makes sense, given our transactions provide atomicity but not isolation guarantees!

Reply:
Yeah, it's kinda weird :/ Though, the transaction changes themselves and the underlying state are still "locked" during the transaction, as you stated differently above. Really I just needed the ability to update the collection after catching a schema mismatch error _during the txn_.